lib/segment/src/segment.rs

```diff
@@ -2071,4 +2049,159 @@ mod tests {
+                        quantization_config: None,
+                    },
+                ),
+                (
+                    "b".into(),
+                    VectorDataConfig {
+                        size: 2,
+                        distance: Distance::Dot,
+                        storage_type: VectorStorageType::Memory,
+                        index: Indexes::Plain {},
+                        quantization_config: None,
+                    },
+                ),
+            ]),
+            payload_storage_type: Default::default(),
+        };
+        let mut segment = build_segment(dir.path(), &config, true).unwrap();
+
+        // Insert one point for a reference internal ID
+        let point_id = 4.into();
+        segment
+            .upsert_point(
+                100,
+                point_id,
+                &NamedVectors::from([
+                    ("a".into(), vec![0.1, 0.2, 0.3, 0.4]),
+                    ("b".into(), vec![1.0, 0.9]),
+                ]),
+            )
+            .unwrap();
+        let internal_id = segment.lookup_internal_id(point_id).unwrap();
+
+        // A set of broken vectors
+        let wrong_vectors_single = vec![
+            // Incorrect dimensionality
+            ("a", vec![]),
+            ("a", vec![0.0, 1.0, 0.0]),
+            ("a", vec![0.0, 1.0, 0.0, 1.0, 0.0]),
+            ("b", vec![]),
+            ("b", vec![0.5]),
+            ("b", vec![0.0, 0.1, 0.2, 0.3]),
+            // Incorrect names
+            ("aa", vec![0.0, 0.1, 0.2, 0.3]),
+            ("bb", vec![0.0, 0.1]),
+        ];
+        let wrong_vectors_multi = vec![
+            // Incorrect dimensionality
+            NamedVectors::from_ref("a", &[]),
+            NamedVectors::from_ref("a", &[0.0, 1.0, 0.0]),
+            NamedVectors::from_ref("a", &[0.0, 1.0, 0.0, 1.0, 0.0]),
+            NamedVectors::from_ref("b", &[]),
+            NamedVectors::from_ref("b", &[0.5]),
+            NamedVectors::from_ref("b", &[0.0, 0.1, 0.2, 0.3]),
+            NamedVectors::from([
+                ("a".into(), vec![0.1, 0.2, 0.3]),
+                ("b".into(), vec![1.0, 0.9]),
+            ]),
+            NamedVectors::from([
+                ("a".into(), vec![0.1, 0.2, 0.3, 0.4]),
+                ("b".into(), vec![1.0, 0.9, 0.0]),
+            ]),
+            // Incorrect names
+            NamedVectors::from_ref("aa", &[0.0, 0.1, 0.2, 0.3]),
+            NamedVectors::from_ref("bb", &[0.0, 0.1]),
+            NamedVectors::from([
+                ("aa".into(), vec![0.1, 0.2, 0.3, 0.4]),
+                ("b".into(), vec![1.0, 0.9]),
+            ]),
+            NamedVectors::from([
+                ("a".into(), vec![0.1, 0.2, 0.3, 0.4]),
+                ("bb".into(), vec![1.0, 0.9]),
+            ]),
+        ];
+        let wrong_names = vec!["aa", "bb", ""];
+
+        for (vector_name, vector) in wrong_vectors_single.iter() {
+            check_vector(vector_name, vector, &config).err().unwrap();
```

Comment:
How about this for all of these checks instead of unwrapping?
```suggestion
            assert!(check_vector(vector_name, vector, &config).is_err());
```